packages/components/src/pages/ChatSearch.vue

```diff
@@ -562,35 +567,51 @@ export default {
 
         const userProvidedContext: ExplainRpc.UserContextItem[] = [];
         if (this.pinnedItems.length > 0) {
-          userProvidedContext.push(
-            ...this.pinnedItems.map((p) => {
-              if (p.type === 'file') {
-                return {
-                  type: 'file',
-                  location: p.location,
-                };
-              } else {
-                return {
-                  type: 'code-snippet',
-                  location: p.location,
-                  content: p.content,
-                };
-              }
-            })
-          );
+          const pinnedItemContextItems = this.pinnedItems.map((p) => {
+            if (p.type === 'file') {
+              return {
+                type: 'file',
+                location: p.location,
+              };
+            } else {
+              return {
+                type: 'code-snippet',
+                location: p.location,
+                content: p.content,
+              };
+            }
+          });
+          userProvidedContext.push(...pinnedItemContextItems);
         }
 
         if (codeSelections.length > 0) {
-          userProvidedContext.push(
-            ...codeSelections.map((c) => ({
+          const codeSelectionContextItems = codeSelections.map((c) => {
+            const result: ExplainRpc.UserContextItem = {
               type: 'code-selection',
               content: c,
```

Comment:
I believe this should be `c.code`? I don't know why this isn't typechecked.

Reply:
Yeah; it should be the content.